sql/core/src/test/scala/org/apache/spark/sql/DataFrameHintSuite.scala

```diff
@@ -68,5 +68,23 @@ class DataFrameHintSuite extends AnalysisTest with SharedSQLContext {
     check(
       df.hint("REPARTITION", 100),
       UnresolvedHint("REPARTITION", Seq(100), df.logicalPlan))
+
+    check(
+      df.hint("REPARTITION", 10, $"id".expr),
+      UnresolvedHint("REPARTITION", Seq(10, $"id".expr), df.logicalPlan))
+
+    val e1 = intercept[AnalysisException] {
```

Comment:
If we want to support column, we should add code at `Dataset.hint()` that convert `Column` to `Expression` just like `Dataset.repartition()` and `Dataset.repartitionByRange()`.